// Copyright 2025 The Bucketeer Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package bucketeer.eventcounter;
option go_package = "github.com/bucketeer-io/bucketeer/proto/eventcounter";

import "proto/eventcounter/variation_result.proto";

message GoalResult {
  string goal_id = 1;
  repeated VariationResult variation_results = 2;
  Summary summary = 3;
}

message Summary {
  message Variation {
    string id = 1;
    double probability = 2;
    bool outperformed = 3;
  }
  repeated Variation best_variations = 1;
  int64 total_evaluation_user_count = 2;
  int64 total_goal_user_count = 3;
}